WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -476,32 +477,52 @@ - (void)preloadStats
 
 - (void)preloadPosts
 {
-    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
-    PostService *postService = [[PostService alloc] initWithManagedObjectContext:context];
-    PostListFilterSettings *filterSettings = [[PostListFilterSettings alloc] initWithBlog:self.blog postType:PostServiceTypePost];
-    PostListFilter *filter = [filterSettings currentPostListFilter];
-    
-    PostServiceSyncOptions *options = [PostServiceSyncOptions new];
-    options.statuses = filter.statuses;
-    options.authorID = [filterSettings authorIDFilter];
-    options.purgesLocalSync = YES;
-    
-    [postService syncPostsOfType:PostServiceTypePost withOptions:options forBlog:self.blog success:nil failure:nil];
+    NSTimeInterval now = [[NSDate date] timeIntervalSinceReferenceDate];
+    NSTimeInterval lastSync = self.blog.lastPostsSync.timeIntervalSinceReferenceDate;
+    if (now - lastSync > PreloadingCacheTimeout) {
+        NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+        PostService *postService = [[PostService alloc] initWithManagedObjectContext:context];
+        PostListFilterSettings *filterSettings = [[PostListFilterSettings alloc] initWithBlog:self.blog postType:PostServiceTypePost];
+        PostListFilter *filter = [filterSettings currentPostListFilter];
+        
+        PostServiceSyncOptions *options = [PostServiceSyncOptions new];
+        options.statuses = filter.statuses;
+        options.authorID = [filterSettings authorIDFilter];
+        options.purgesLocalSync = YES;
+
+        self.blog.lastPostsSync = [NSDate date];
+        NSError *error = nil;
+        [self.blog.managedObjectContext save:&error];
+
+        [postService syncPostsOfType:PostServiceTypePost withOptions:options forBlog:self.blog success:nil failure:^(NSError *error) {
+            self.blog.lastPostsSync = [NSDate dateWithTimeIntervalSince1970:0.0]; // invalidate cache timer
+        }];
+    }
 }
 
 - (void)preloadPages
```

Comment:
Since `preloadPages` and `preloadPosts` are nearly identical, can we move the bulk of these methods to one shared method that accepts the type and whatever else is unique? We could still have both `preloadPages` and `preloadPosts` as helper methods pointing to another shared method.